### PR TITLE
Home work 22

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: src/src/main/java/controller/ChangeProductServlet.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.logging.Logger;
+
+@WebServlet(value = "/users/change")
```
Comment (Sofasmile) [inline src/src/main/java/controller/ChangeProductServlet.java]: better rename to /users/edit

File: src/src/main/java/controller/ChangeProductServlet.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+public class ChangeProductServlet extends HttpServlet {
+
+    private static final Logger LOGGER = Logger.getLogger(String.valueOf(ChangeProductServlet.class));
+    private static final ProductService productService = ProductServiceFactory.getInstance();
+    private Product product;
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
```
Comment (Sofasmile) [inline src/src/main/java/controller/ChangeProductServlet.java]: https://mate-academy.github.io/style-guides/java/java.html#s4.4-column-limit

File: src/src/main/java/controller/ProductServlet.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(name = "/products")
+public class ProductServlet extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        request.getRequestDispatcher("products.jsp").forward(request, response);
```
Comment (Sofasmile) [inline src/src/main/java/controller/ProductServlet.java:25]: better use method productService.getAll() here

File: src/src/main/java/dao/impl/UsersDAOImpl.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Override
+    public List getAll() {
+        return DataBase.users;
+    }
+
+    @Override
+    public User getUserById(Long userId) {
+        return DataBase.users.stream().filter(e -> false).findFirst().get();
```
Comment (Sofasmile) [inline src/src/main/java/dao/impl/UsersDAOImpl.java]: what does this method do?

File: src/src/main/java/dao/impl/UsersDAOImpl.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Override
+    public User getUserByEmail(String email) {
+        return DataBase.users.stream()
+                .filter(e -> e.getEmail().equals(email))
+                .findFirst()
+                .orElse(new User("admin", "admin", ""));
```
Comment (Sofasmile) [inline src/src/main/java/dao/impl/UsersDAOImpl.java]: we can initialize userList with default user

File: src/src/main/java/factories/ProductServiceFactory.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import service.Impl.ProductServiceImpl;
+import service.ProductService;
+
+public class ProductServiceFactory {
+    private static ProductService instance;
+
+    private ProductServiceFactory() {
+    }
+
+    public static synchronized ProductService getInstance() {
```
Comment (Sofasmile) [inline src/src/main/java/factories/ProductServiceFactory.java:13]: why we use word 'synchronized' in this class and don't use in class UserDAOFactory?

File: src/src/main/java/service/Impl/UserServiceInpl.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
+package service.Impl;
+
+import dao.UsersDAO;
+import factories.UserDAOFactory;
+import model.User;
+import service.UserService;
+
+import java.util.List;
+
+public class UserServiceInpl implements UserService<User> {
```
Comment (Sofasmile) [inline src/src/main/java/service/Impl/UserServiceInpl.java:10]:
```suggestion
public class UserServiceImpl implements UserService<User> {
```

Comment (UnitOmega13) [reply src/src/main/java/service/Impl/UserServiceInpl.java:10]: fixed

File: src/src/main/java/controller/ChangeProductServlet.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.logging.Logger;
+
+@WebServlet(value = "/users/edit")
+public class ChangeProductServlet extends HttpServlet {
+
+    private static final Logger LOGGER = Logger.getLogger(String.valueOf(ChangeProductServlet.class));
+    private static final ProductService productService = ProductServiceFactory.getInstance();
+    private Product product;
```
Comment (Sofasmile) [inline src/src/main/java/controller/ChangeProductServlet.java]: try doesn't use private field to edit product

File: src/src/main/java/controller/ChangeUserServlet.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/products/edit")
+public class ChangeUserServlet extends HttpServlet {
+
+    private static final Logger LOGGER = Logger.getLogger(String.valueOf(ChangeProductServlet.class));
+    private static final UserService userService = UserServiceFactory.getInstance();
+    private User user;
```
Comment (Sofasmile) [inline src/src/main/java/controller/ChangeUserServlet.java]: the same

File: src/src/main/java/controller/SignUpServlet.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/registration")
+public class SignUpServlet extends HttpServlet {
+
+    private static final UserService USER_SERVICE = UserServiceFactory.getInstance();
```
Comment (Sofasmile) [inline src/src/main/java/controller/SignUpServlet.java]: camelCase please

**Pull request closed without merging**